gpx_test.go

```diff
@@ -611,19 +613,29 @@ func TestTime(t *testing.T) {
 }
 
 func TestParseExamples(t *testing.T) {
-	for _, filename := range []string{
-		"testdata/ashland.gpx",
-		"testdata/fells_loop.gpx",
-		"testdata/mystic_basin_trail.gpx",
-	} {
-		t.Run(filename, func(t *testing.T) {
-			f, err := os.Open(filename)
-			assert.NoError(t, err)
-			defer f.Close()
-			_, err = gpx.Read(f)
-			assert.NoError(t, err)
-		})
-	}
+	err := filepath.WalkDir("testdata", func(filename string, d fs.DirEntry, errs error) error {
```

Comment:
As requested before, please use [`fs.WalkDir`](https://pkg.go.dev/io/fs#WalkDir). You can use [`os.DirFS("testdata")`](https://pkg.go.dev/os#DirFS) to get the `fs.FS` argument.

Reply:
Changed